_posts/2018-10-17-why-ruby-on-rails.md

```diff
@@ -0,0 +1,151 @@
+- Weird structure - no semi-colon to end a statement, optional braces in function calls, no curly brackets in function definitions, and more.
+
+For a relatively quick rundown of ruby syntax, the [ruby koans](http://rubykoans.com/) can be a useful resource.
+The koans are a test driven walk through the ruby language that aims to teach syntax, structure and basic functions.
+Once you reach *enlightenment* you'll be ready to not get overwhelmed by rails.
+You can also find the official documentation at [ruby-doc](https://ruby-doc.org/).
+
+### Installation
+
+If you're working on a linux based system, ruby will be available from the repositories. On debian/ubuntu based systems:
+   
+    $ sudo apt-get install ruby
+
+This will install ruby system-wide for all users. There are a few issues with installing ruby this way. 
+First if on an older distribution, the installed ruby packages may be outdated.
+Second, what happens if you want to work on a project that relies on a lower ruby version?
+You would have to downgrade your install, work on the project, and when done re-install the higher version. Not very performant.
+
+A ruby version manager helps smooth out these bumps. Popular options include [rbenv] and [RVM]. I've used RVM so it's what I'd recommend.
+RVM is installed on a per-user basis and installs the different ruby versions in a folder in the user's home directory.
+Follow instructions on the [install page](https://rvm.io/rvm/install) to get rvm.
+Once installed you simply choose which ruby version to develop on:
+
+    $ rvm list known # show available rubies
+    $ rvm install 2.5.1
+    $ rvm use 2.5.1 --default # use ruby-2.5.1 and set it as default
+    $ ruby -v # make sure that you have ruby installed
+
+With ruby installed we're ready to install rails:
+
+    $ gem -v # ensure that you have rubygems installed
+    $ gem install bundler # dependency management package
+    $ gem install rails # fetch the latest rails packages
+    $ rails -v # show installed rails version
+
+> Always ensure that the rvm PATH entry is the last line in your .bashrc
+
+### Getting Started
+
+The [official documentation](https://guides.rubyonrails.org/getting_started.html) does a great job of getting you started.
+It walks you through the process of developing a weblog, with a stepwise introduction to the various features of rails.
+
+The [Rails Girls](http://guides.railsgirls.com/) website also has a great collection of resources.
+The guides take you from installation to building your apps and through to testing and deployment.
+And no, the resources aren't just meant for girls.
+
+If, like me, getting through an entire guide totally drains you, I'd recommend going through just enough material to get your own project started.
+The guides can then serve as reference material. After all, not all of us want to write blogs or need another todo app.
+
+### Common Rails Commands
+
+    $ rails new <appname> # start a new rails project
```

Comment:
Use comments to explain alternative syntax. Someone copying and pasting this code will wonder why it does not work. like so
```bash
rails s # or rails server => Starts the server
```
or omit the alternative  for simplicity